pkg/operator/controller/dns/controller.go

```diff
@@ -130,33 +141,32 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	var statuses []iov1.DNSZoneStatus
 	for i := range zones {
 		zone := zones[i]
-		var conditions []iov1.DNSZoneCondition
-		err := r.dnsProvider.Ensure(record, zone)
-		if err != nil {
-			conditions = append(conditions, iov1.DNSZoneCondition{
-				Type:               iov1.DNSRecordFailedConditionType,
-				Status:             string(operatorv1.ConditionTrue),
-				Reason:             "ProviderError",
-				Message:            fmt.Sprintf("The DNS provider failed to ensure the record: %v", err),
-				LastTransitionTime: metav1.Now(),
-			})
+		condition := iov1.DNSZoneCondition{
+			Status:             string(operatorv1.ConditionUnknown),
+			Type:               iov1.DNSRecordFailedConditionType,
+			LastTransitionTime: metav1.Now(),
+		}
+
+		if len(zone.ID) == 0 {
+			condition.Status = string(operatorv1.ConditionTrue)
+			condition.Reason = "EmptyZoneId"
+			condition.Message = "Zone has an empty id"
```

Comment:
Actually, it is valid for the zone id to be empty if the zone is specified by tag.  For example, see `getZoneID` in `pkg/dns/aws/dns.go`:
```go
// getZoneID finds the ID of given zoneConfig in Route53. If an ID is already
// known, return that; otherwise, use tags to search for the zone. Returns an
// error if the zone can't be found.
```
And the cluster DNS configuration:
```yaml
  spec:
    baseDomain: ci-op-8xt2f72z-0e76d.origin-ci-int-aws.dev.rhcloud.com
    privateZone:
      tags:
        Name: ci-op-8xt2f72z-0e76d-rvgwz-int
        kubernetes.io/cluster/ci-op-8xt2f72z-0e76d-rvgwz: owned
    publicZone:
      id: Z2GYOLTZHS5VK
```
CI is failing because the operator is not publishing records for the private zone, which is configured with tags, not the zone id:
```
status:
  zones:
  - conditions:
    - lastTransitionTime: "2020-04-01T03:44:21Z"
      message: Zone has an empty id
      reason: EmptyZoneId
      status: "True"
      type: Failed
    dnsZone:
      tags:
        Name: ci-op-8xt2f72z-0e76d-rvgwz-int
        kubernetes.io/cluster/ci-op-8xt2f72z-0e76d-rvgwz: owned
  - conditions:
    - lastTransitionTime: "2020-04-01T03:44:21Z"
      message: The DNS provider succeeded in ensuring the record
      reason: ProviderSuccess
      status: "False"
      type: Failed
    dnsZone:
      id: Z2GYOLTZHS5VK
```
We should delete this check for an empty zone.  (Alternatively, we could report an error if neither id nor tags are specified.)